yt/yt/server/master/node_tracker_server/node_tracker.cpp

```diff
@@ -1170,6 +1201,61 @@ class TNodeTracker
         }
     }
 
+    void CreateDataCenterObject(TNode* node, const TString& Name)
+    {
+        YT_VERIFY(HasMutationContext());
+        YT_VERIFY(Bootstrap_->IsPrimaryMaster());
+
+        auto req = TMasterYPathProxy::CreateObject();
+        req->set_type(static_cast<int>(EObjectType::DataCenter));
+
+        auto attributes = CreateEphemeralAttributes();
+        attributes->Set("name", Name);
+        ToProto(req->mutable_object_attributes(), *attributes);
+
+        const auto& rootService = Bootstrap_->GetObjectManager()->GetRootService();
+        try {
+            SyncExecuteVerb(rootService, req);
+        } catch (const std::exception& ex) {
+            YT_LOG_ALERT(ex, "Failed to create data_center for a node");
+
+            if (IsObjectAlive(node)) {
```

Comment:
I am not sure, but shouldn't we also unref the rack in this case, as it is also going to be dangling?

Reply:
cc @gritukan

Reply:
No, it's fine